lightning/src/chain/keysinterface.rs

```diff
@@ -226,7 +226,7 @@ impl Readable for SpendableOutputDescriptor {
 /// of LN security model, orthogonal of key management issues.
 // TODO: We should remove Clone by instead requesting a new Sign copy when we create
 // ChannelMonitors instead of expecting to clone the one out of the Channel into the monitors.
-pub trait Sign : Send+Clone + Writeable {
+pub trait BaseSign : Send + Writeable {
```

Comment:
`BaseSign` seems like a code smell to me. Can't we define `Sign` like this and then use `Sign + Clone` where needed?

Reply:
We had it that way, but @TheBlueMatt had some concerns about complexity / changes to bindings.  It's about 20 spots in the code if I remember correctly.

Reply:
Gotcha. Figured there might something related to binding. We can remove it easily in the future, so no objections keeping it as is.